config/common/common.go

```diff
@@ -93,3 +103,102 @@ func GetFullyQualifiedIDFn(serviceProvider string, keyPairs ...string) tjconfig.
 		return path, nil
 	}
 }
+
+var referenceRules = [][]string{
+	{"resource_group$", "/azure/v1alpha1.ResourceGroup"},
+	{"subnet$", "/network/v1alpha1.Subnet"},
+}
+
+// AddCommonReferences adds some common reference fields.
+// This is a part of resource generation pipeline.
+func AddCommonReferences(references tjconfig.References, resource *schema.Resource, shortGroup string, nestedFieldNames []string) error {
```

Comment:
nit: We may remove the `nestedFieldNames` from the signature to make the initial call simpler. And add a recursive unexported function `addCommonReferences` with the `nestedFieldNames` parameter in its signature, which is called by the `AddCommonReferences` function, with the initial empty value. 

```suggestion
func AddCommonReferences(r *config.Reference) error {
   return addCommonReferences(r *config.Reference, []string{})
}
```